File: Python/underautomation/universal_robots/primary_interface/cartesian_info_package_event_args.py
import typing
from underautomation.universal_robots.common.package_event_args import PackageEventArgs
import clr
import os
clr.AddReference(os.path.realpath(os.path.join(os.path.dirname(__file__), "..",  'lib', 'UnderAutomation.UniversalRobots.dll')))
from UnderAutomation.UniversalRobots.PrimaryInterface import CartesianInfoPackageEventArgs as cartesian_info_package_event_args

class CartesianInfoPackageEventArgs(PackageEventArgs):
	def __init__(self, _internal = 0):
		if(_internal == 0):
			self._instance = cartesian_info_package_event_args()
		else:
			self._instance = _internal
	@property
	def x(self) -> float:
		return self._instance.X
	@x.setter
	def x(self, value: float):
		self._instance.X = value
	@property
	def y(self) -> float:
		return self._instance.Y
	@y.setter
	def y(self, value: float):
		self._instance.Y = value
	@property
	def z(self) -> float:
		return self._instance.Z
	@z.setter
	def z(self, value: float):
		self._instance.Z = value
	@property
	def rx(self) -> float:
		return self._instance.Rx
	@rx.setter
	def rx(self, value: float):
		self._instance.Rx = value
	@property
	def ry(self) -> float:
		return self._instance.Ry
	@ry.setter
	def ry(self, value: float):
		self._instance.Ry = value
	@property
	def rz(self) -> float:
		return self._instance.Rz
	@rz.setter
	def rz(self, value: float):
		self._instance.Rz = value
	@property
	def tcp_offset_x(self) -> float:
		return self._instance.TCPOffsetX
	@tcp_offset_x.setter
	def tcp_offset_x(self, value: float):
		self._instance.TCPOffsetX = value
	@property
	def tcp_offset_y(self) -> float:
		return self._instance.TCPOffsetY
	@tcp_offset_y.setter
	def tcp_offset_y(self, value: float):
		self._instance.TCPOffsetY = value
	@property
	def tcp_offset_z(self) -> float:
		return self._instance.TCPOffsetZ
	@tcp_offset_z.setter
	def tcp_offset_z(self, value: float):
		self._instance.TCPOffsetZ = value
	@property
	def tcp_offset_rx(self) -> float:
		return self._instance.TCPOffsetRX
	@tcp_offset_rx.setter
	def tcp_offset_rx(self, value: float):
		self._instance.TCPOffsetRX = value
	@property
	def tcp_offset_ry(self) -> float:
		return self._instance.TCPOffsetRY
	@tcp_offset_ry.setter
	def tcp_offset_ry(self, value: float):
		self._instance.TCPOffsetRY = value
	@property
	def tcp_offset_rz(self) -> float:
		return self._instance.TCPOffsetRZ
	@tcp_offset_rz.setter
	def tcp_offset_rz(self, value: float):
		self._instance.TCPOffsetRZ = value

File: UnderAutomation.UniversalRobots.ObfuscatedSources/PrimaryInterface/CartesianInfoPackageEventArgs.cs
//              WARNING
// This file is an empty shell containing only public C# items.
// The internal code is hidden; to access it, you need to obtain a Source licence of the library.

using Common;

namespace PrimaryInterface {
	/// <summary>
	/// Contains current cartesian position of the robot, including its TCP offset
	/// </summary>
	public class CartesianInfoPackageEventArgs : PackageEventArgs {

		/// <summary>
		/// X axis coordinate in meter of the TCP in the current frame
		/// </summary>
		[PackageDescription("X axis coordinate in meter of the TCP in the current frame", PackageUnit.Meter)]
public double X { get; set; }

		/// <summary>
		/// Y axis coordinate in meter of the TCP in the current frame
		/// </summary>
		[PackageDescription("Y axis coordinate in meter of the TCP in the current frame", PackageUnit.Meter)]
public double Y { get; set; }

		/// <summary>
		/// Z axis coordinate in meter of the TCP in the current frame
		/// </summary>
		[PackageDescription("Z axis coordinate in meter of the TCP in the current frame", PackageUnit.Meter)]
public double Z { get; set; }

		/// <summary>
		/// RX axis coordinate in rad of the TCP in the current frame
		/// </summary>
		[PackageDescription("RX axis coordinate in rad of the TCP in the current frame", PackageUnit.Radian)]
public double Rx { get; set; }

		/// <summary>
		/// RY axis coordinate in rad of the TCP in the current frame
		/// </summary>
		[PackageDescription("RY axis coordinate in rad of the TCP in the current frame", PackageUnit.Radian)]
public double Ry { get; set; }

		/// <summary>
		/// RZ axis coordinate in rad of the TCP in the current frame
		/// </summary>
		[PackageDescription("RZ axis coordinate in rad of the TCP in the current frame", PackageUnit.Radian)]
public double Rz { get; set; }

		/// <summary>
		/// X position of the TCP in the flange frame in meter
		/// </summary>
		[PackageDescription("X position of the TCP in the flange frame in meter", PackageUnit.Meter)]
public double TCPOffsetX { get; set; }

		/// <summary>
		/// Y position of the TCP in the flange frame in meter
		/// </summary>
		[PackageDescription("Y position of the TCP in the flange frame in meter", PackageUnit.Meter)]
public double TCPOffsetY { get; set; }

		/// <summary>
		/// Z position of the TCP in the flange frame in meter
		/// </summary>
		[PackageDescription("Z position of the TCP in the flange frame in meter", PackageUnit.Meter)]
public double TCPOffsetZ { get; set; }

		/// <summary>
		/// RX position of the TCP in the flange frame in rad
		/// </summary>
		[PackageDescription("RX position of the TCP in the flange frame in rad", PackageUnit.Radian)]
public double TCPOffsetRX { get; set; }

		/// <summary>
		/// RY position of the TCP in the flange frame in rad
		/// </summary>
		[PackageDescription("RY position of the TCP in the flange frame in rad", PackageUnit.Radian)]
public double TCPOffsetRY { get; set; }

		/// <summary>
		/// RZ position of the TCP in the flange frame in rad
		/// </summary>
		[PackageDescription("RZ position of the TCP in the flange frame in rad", PackageUnit.Radian)]
public double TCPOffsetRZ { get; set; }
	}
}
